docs/CharmsInDetail.md

```diff
@@ -0,0 +1,292 @@
+# Charms in detail
+## Table of contents
+
+[Best practices](##Best%20practice)
+
+[Charm writing in detail](##Charm%20Writing%20in%20detail)
+
+[New to charms?](##New%20to%20charms?)
+
+[Charm Writing](#Charm%20writing)
+
+[Example Charms](##Example%20Charms)
+
+## Best practices
+
+### Repoository Naming
+
+Naming follows the convention of charm-<charm_name> for example `charm-percona-cluster`.
+
+The rationale behind this is quick and easy searching of the 1000s of github projects along with other reasons.
+
+
+## Charm Writing in detail
+
+## New to charms?
+
+> Here are some completely reasonable questions to be asking right now
+
+### Why charms?
+
+Charms have a noble goal: take the complication of managing 100s potentially 1000s of configuration files and abstract them as python modules. Thus by abstraction creating a true insfrastructure as code.
+
+But charms are not just limited to setting up configuration, they also handle the installation of charms, and can implement actions thus providing not just configuration, but day zero to day three operations support.
```

Comment:
Is "day three" really a thing now? I'm honestly just curious because I only know of day 0, 1, 2, and EOL: https://dzone.com/articles/defining-day-2-operations

Reply:
that would be my half-context switch from 0 base to 1 base.